src/shared/hooks.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useRef } from "react";
+
+export function useDebouncedCallback<A extends any[]>(
```

Comment:
Should this be `export const ...` for consistency?

Reply:
Full disclosure: I copied and pasted this from StackOverflow.

Reply:
Sorry, I replied to this via email but it turns out GitHub just creates a new comment instead.

It's advisable to link the the SO answer that you got this from, then. They license snippets under CC-BY SA 4.0 so attribution is required.